numba_cuda/numba/cuda/lowering.py

```diff
@@ -0,0 +1,41 @@
+from numba.core.lowering import Lower
+from llvmlite import ir
+
+
+class CUDALower(Lower):
+    def storevar(self, value, name, argidx=None):
+        """
+        Store the value into the given variable.
+        """
+        super().storevar(value, name, argidx)
+
+        # Emit llvm.dbg.value instead of llvm.dbg.declare for local scalar
+        # variables immediately after a store instruction.
+        if (
+            self.context.enable_debuginfo
+            and (
+                name not in self._singly_assigned_vars
+                or self._disable_sroa_like_opt
+            )
+            and not name.startswith("$")
```

Comment:
Perhaps the following clarifies these for other readers:
```suggestion
            # Conditions used to elide stores in parent method
            and (
                name not in self._singly_assigned_vars
                or self._disable_sroa_like_opt
            )
            # No emission of debuginfo for internal names
            and not name.startswith("$")
```

Reply:
Yes, I will apply the suggested comments.